pandas/_libs/writers.pyx

```diff
@@ -144,7 +144,7 @@ cpdef inline Py_ssize_t word_len(object val):
         Py_ssize_t l = 0
 
     if isinstance(val, str):
-        l = PyUnicode_GET_SIZE(val)
+        l = PyUnicode_GET_LENGTH(val)
```

Comment:
i think if u import this only for 3.9 (or maybe 3.8) it will work

there are examples of this in the cython code i think or look at the cython docs on how to do this

Reply:
this must be new c api

Reply:
Those two failed builds have cython version pinned to 0.29.16, bump it to 0.29.21 fixed it.